packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { VirtualizerMeasureProps } from './useVirtualizerMeasure.types';
+import { debounce } from '../utilities/debounce';
+
+/**
+ * React hook that measures virtualized space based on a static size to ensure optimized virtualization length.
+ */
+export const useStaticVirtualizerMeasure = (
+  virtualizerProps: VirtualizerMeasureProps,
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+  scrollRef: (instance: HTMLElement | HTMLDivElement | null) => void;
+} => {
+  const { defaultItemSize, direction = 'vertical' } = virtualizerProps;
+
+  const [state, setState] = React.useState({
+    virtualizerLength: 0,
+    bufferSize: 0,
+    bufferItems: 0,
+  });
+
+  const { virtualizerLength, bufferItems, bufferSize } = state;
+
+  // The ref the user sets on their scrollView - Defaults to document.body to ensure no null on init
+  const container: React.MutableRefObject<HTMLElement | null> = React.useRef<HTMLElement | null>(null);
```

Comment:
```suggestion
  const container = React.useRef<HTMLElement | null>(null);
```

nit: I don't think you need this typing